dataService.js

```diff
@@ -0,0 +1,53 @@
+// Getting data from json files
+const categoryData = require('./src/data/categories.json');
+const productData = require('./src/data/products.json');
+
+// Maps
+let productMap = {};
+let categoryMap = {};
+let combinedProductMap = [];
+
+function getProducts() {
+  productData.products.forEach(product => {
+    productMap[product.id] = product;
+  });
+}
+
+function getCategories() {
+  categoryData.categories.forEach(category => {
+    categoryMap[category.id] = category.categoryName;
+  });
+}
+
+function combineProductsWithCategories() {
+
+  // Creating map of categories
+  getCategories();
```

Comment:
No need to create it alway. We can send if we already combined

Reply:
Don't we have `getCombinedProductMap()` method to do that?

Reply:
We don't need to call this methods and steps always. If we already combined, and assigend to `combinedProductMap`, next time - we can simply return that

Reply:
Have a look at [this line](https://github.com/PrathameshMutkure/my-app/blob/dev/dataService.js#L79) in which we have `getCombinedProductMap()` method which does exactly that, check if already combined and work accordingly. I'm not exporting this method, the user will have to always us `getCombinedProductMap()` which does exactly what you pointed out.

I'm sorry if I'm missing anything. Thank You :)